packages/client/src/generation/generateClient.ts

```diff
@@ -124,12 +123,38 @@ export async function buildClient({
     edge: true,
   })
 
+  const trampolineTsClient = new TSClient({
+    ...nodeClientOptions,
+    reusedTs: 'index',
+    reusedJs: '#main-entry-point',
+  })
```

Comment:
This is the core surface of the change. When `require('#main-entry-point')` is executed, it goes through an `imports` map.